File: Programmers/[12904] 가장 긴 팰린드롬/ONE/12904.java
class Solution {
    public int solution(String s) {
        int n = s.length();
        int answer = 1;

        for (int i = 0; i < n - 1; i++) {
            int len = 1;
            int left = i - 1, right = i + 1;

            answer = Math.max(answer, getAnswer(s, n, len, left, right));

            if (s.charAt(i) == s.charAt(i + 1)) {
                len = 2;
                left = i - 1;
                right = i + 2;

                answer = Math.max(answer, getAnswer(s, n, len, left, right));
            }
        }
        return answer;
    }

    private int getAnswer(String s, int n, int len, int left, int right) {
        while (left >= 0 && right < n) {
            if (s.charAt(left) != s.charAt(right))
                break;
            len += 2;
            left -= 1;
            right += 1;
        }
        return len;
    }
}

File: Programmers/[12904] 가장 긴 팰린드롬/JFe/12904.py
def solution(s):
    answer = 0
    for i in range(len(s)):
        # 팰린드롬 문자열이 홀수인 경우
        left, right, count = i - 1, i + 1, 1
        while 0 <= left and right < len(s):
            if s[left] == s[right]:
                count += 2
                left, right = left - 1, right + 1
            else: break
        answer = max(answer, count)
        # 팰린드롬 문자열이 짝수인 경우
        pivot, right, count = i, i + 1, 0
        while 0 <= pivot and right < len(s):
            if s[pivot] == s[right]:
                count += 2
                pivot, right = pivot - 1, right + 1
            else: break
        answer = max(answer, count)
    return answer